### PR TITLE
set up associations

### PR DESCRIPTION
**Pull request by devMunyi**
1) Created a class Classroom with the following:

- **label** instance variable, initialized in the constructor.
- Setter and getter for **label**(used attr_accessor).

2) Created the has-many/belongs-to relationship between Classroom and Student. The following was implemented:
- created the has-many side (a classroom has many students).
- created the belongs-to side (a student belongs to a classroom).
- ensured that when adding a student to a classroom it also sets the classroom for the student.
- ensured that when setting the classroom for a student it also adds it to the classrooms' students.

3) Created a class Book with the following:

- @title and @author instance variables initialized in the constructor.
- Setters and getters for instance variables (used attr_accessor).

4) Created a class Rental with the following:

- **date** instance variable initialized in the constructor.
- Setter and getter for **date** (used attr_accessor).

5) Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following was implemented:

- Created the has-many side of Book and Rental (a book has many rentals).
- Created the belongs-to side of Rental and Book (a rental belongs to a book).
- Created the has-many side of Person and Rental (a person has many rentals).
- Created the belongs-to side of Rental and Person (a rental belongs to a person).
- Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, person
```
Comment (m05e5) [inline rental.rb]: - [ ]  Getters and Setters for @book and @person instance variables should be present. Kindly change from `attr_reader` to `attr_accessor` and also fix the `person` to `:person`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (m05e5) [inline book.rb]: - [ ] Hi as part of the requirements, you are supposed to do a has many relations between classroom and students. kindly add the method that serves as the has-many relation in books

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @devMunyi , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Please check the comments below 

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ha-manel — APPROVED** [review]
# STATUS: APPROVED ✅

## Hi @devMunyi 👋

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: 
**Congratulations! 🎉**



## Highlights:
* Well-written PR description ✔️ 
* All tests passed ✔️ 
* Associations are set up correctly 💯 

## Optional suggestions:
- [ ] Please consider capitalizing your commit messages to follow best practices 👍 


------
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**